src/main/java/cloud/orbit/spring/service/OrbitServiceConfiguration.java

```diff
@@ -0,0 +1,87 @@
+/*
+ Copyright (C) 2017 Electronic Arts Inc.  All rights reserved.
+
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions
+ are met:
+
+ 1.  Redistributions of source code must retain the above copyright
+     notice, this list of conditions and the following disclaimer.
+ 2.  Redistributions in binary form must reproduce the above copyright
+     notice, this list of conditions and the following disclaimer in the
+     documentation and/or other materials provided with the distribution.
+ 3.  Neither the name of Electronic Arts, Inc. ("EA") nor the names of
+     its contributors may be used to endorse or promote products derived
+     from this software without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY ELECTRONIC ARTS AND ITS CONTRIBUTORS "AS IS" AND ANY
+ EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ DISCLAIMED. IN NO EVENT SHALL ELECTRONIC ARTS OR ITS CONTRIBUTORS BE LIABLE FOR ANY
+ DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package cloud.orbit.spring.service;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import cloud.orbit.actors.runtime.ActorRuntime;
+import cloud.orbit.actors.runtime.BasicRuntime;
+import cloud.orbit.actors.runtime.DefaultDescriptorFactory;
+import cloud.orbit.actors.runtime.DescriptorFactory;
+
+/**
+ * Register commonly used static helpers and singletons in Orbit as Spring beans
+ */
+@Configuration
+public class OrbitServiceConfiguration
+{
+    @Bean
+    @ConditionalOnMissingBean(ActorReferenceService.class)
+    public ActorReferenceService actorReferenceService()
+    {
+        return new ActorReferenceService();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(RemoteReferenceService.class)
+    public RemoteReferenceService remoteReferenceService()
+    {
+        return new RemoteReferenceService();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(AsyncStreamService.class)
+    public AsyncStreamService asyncStreamService()
+    {
+        return new AsyncStreamService();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(ActorRuntime.class)
+    public ActorRuntime actorRuntime()
+    {
+        return ActorRuntime.getRuntime();
```

Comment:
I have concerns with this and the next two beans. The other Service classes make sense as they don't change the contract of those methods.

This bean construction is asking for a runtime but not allowing for the getRuntime() to be called again within the application to potentially get an new runtime instance. @JoeHegarty is this a big concern? At what time in the Stage lifecycle should the runtime be retrieved? Do we really want to be exposing the Actor and Basic runtime in this manner? How about the Default DescriptorFactory?